curriculum/challenges/english/25-front-end-development/quiz-recursion/66edd43cded6bff30944b676.md

```diff
@@ -17,439 +17,530 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
 #### --answer--
 
-Placeholder answer
+JavaScript
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What's wrong with this recursive DOM traversal?
+
+```javascript
+function findElement(element) {
+    if (element.id === 'target') return element;
+    findElement(element.firstChild);
+}
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+Missing base case.
 
 ---
 
-Placeholder distractor 2
+Wrong parameter.
 
 ---
 
-Placeholder distractor 3
+Incorrect comparison.
 
 #### --answer--
 
-Placeholder answer
+Missing return for recursive call.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+When using recursion, which statement is true?
 
 #### --distractors--
 
-Placeholder distractor 1
+The function always runs infinitely.
 
 ---
 
-Placeholder distractor 2
+The function uses a while loop.
 
 ---
 
-Placeholder distractor 3
+The function has no parameters.
 
 #### --answer--
 
-Placeholder answer
+The function calls itself.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What will this return?
+
+```javascript
+function join(arr) {
+    if (arr.length <= 1) return arr;
+    return [arr[0]].concat(join(arr.slice(1)));
+}
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+A reversed array.
 
 ---
 
-Placeholder distractor 2
+The first element only.
 
 ---
 
-Placeholder distractor 3
+An empty array.
 
 #### --answer--
 
-Placeholder answer
+A copy of the original array.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What technique can improve recursive performance?
 
 #### --distractors--
 
-Placeholder distractor 1
+Using more base cases.
 
 ---
 
-Placeholder distractor 2
+Adding more parameters.
 
 ---
 
-Placeholder distractor 3
+Using global variables.
 
 #### --answer--
 
-Placeholder answer
+Memoization
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What will this recursive function do?
+
+```javascript
+function repeat(str) {
+    return str + repeat(str);
+}
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+Return the string twice.
 
 ---
 
-Placeholder distractor 2
+Return undefined.
 
 ---
 
-Placeholder distractor 3
+Create an empty string.
 
 #### --answer--
 
-Placeholder answer
+Cause a stack overflow.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What HTML structure is typically traversed recursively?
 
 #### --distractors--
 
-Placeholder distractor 1
+Style sheets
 
 ---
 
-Placeholder distractor 2
+Meta tags
 
 ---
 
-Placeholder distractor 3
+Script tags
 
 #### --answer--
 
-Placeholder answer
+The DOM tree
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+In JavaScript, what keyword can help prevent infinite recursion?
 
 #### --distractors--
 
-Placeholder distractor 1
+`loop`
 
 ---
 
-Placeholder distractor 2
+`console`
 
 ---
 
-Placeholder distractor 3
+`try`
 
 #### --answer--
 
-Placeholder answer
+`if`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What does this recursive function do?
```

Comment:
the name of the function gives it away. If you want the camper to interpret the function gives it a generic name please